src/devices/GoPiGo3/Movements/Vehicle.cs

```diff
@@ -0,0 +1,276 @@
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void Backward(int speed)
+        {
+            StartMotor(PortLeft, speed * _correctedDir);
+            StartMotor(PortRight, speed * _correctedDir);
+        }
+
+        /// <summary>
+        /// Run forward at the specified speed
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void Forward(int speed)
+        {
+            Backward(-speed);
+        }
+
+        /// <summary>
+        /// Turn the vehicule left by the specified number of degrees for each motor. So 360 will do 1 motor turn.
+        /// You need to do some math to have the actual vehicule turning fully at 360. It depends of the reduction used.
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="degrees">degrees to turn each motor</param>
+        public void TurnLeft(int speed, int degrees)
+        {
+            RunMotorSyncDegrees(new MotorPort[2] { PortLeft, PortRight }, new int[2] { -speed * _correctedDir, speed * _correctedDir }, new int[2] { degrees, degrees });
+        }
+
+        /// <summary>
+        /// Turn the vehicule right by the specified number of degrees for each motor. So 360 will do 1 motor turn.
+        /// You need to do some math to have the actual vehicule turning fully at 360. It depends of the reduction used.
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="degrees">degrees to turn each motor</param>
+        public void TurnRight(int speed, int degrees)
+        {
+            RunMotorSyncDegrees(new MotorPort[2] { PortLeft, PortRight }, new int[2] { speed * _correctedDir, -speed * _correctedDir }, new int[2] { degrees, degrees });
+        }
+
+        /// <summary>
+        /// Turn the vehicule left for a number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">number of milliseconds to run the motors</param>
+        public void TrunLeftTime(int speed, int timeout)
+        {
+            RunMotorSyncTime(new MotorPort[2] { PortLeft, PortRight }, new int[2] { -speed * _correctedDir, speed * _correctedDir }, timeout);
+        }
+
+        /// <summary>
+        /// Turn the vehicule right for a number of milliseconds
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        /// <param name="timeout">number of milliseconds to run the motors</param>
+        public void TrunRightTime(int speed, int timeout)
+        {
+            RunMotorSyncTime(new MotorPort[2] { PortLeft, PortRight }, new int[2] { speed * _correctedDir, -speed * _correctedDir }, timeout);
+        }
+
+        /// <summary>
+        /// Stop the vehicule
+        /// </summary>
+        public void Stop()
```

Comment:
Lol, I've made almost identical API for my Vehicle class over the weekend (still prototyping though - in my case it wasn't taking any arguments and it was more like a "button pressed" signal which worked better for the real time control, the speed was adjusted with a seperate controller thread)

Reply:
We should definitely work on some common interface for motor classes!